File: abc_spec/code_samples/C#/webhooks@{id}/patch.cs
// For more information please refer to https://github.com/checkout/checkout-sdk-net
using Checkout.Webhooks;

ICheckoutApi api = CheckoutSdk.DefaultSdk().StaticKeys()
    .PublicKey("public_key")
    .SecretKey("secret_key")
    .Environment(Environment.Sandbox)
    .HttpClientFactory(new DefaultHttpClientFactory())
    .Build();

WebhookRequest request = new WebhookRequest()
{
    Url = "https://example.com/webhooks",
    Active = true,
    Headers = new Dictionary<string, string>()
    {
        {"authorization", "1234"}
    },
    ContentType = WebhookContentType.Json,
    EventTypes = new List<string>()
    {
        "payment_approved",
        "payment_pending",
        "payment_declined",
        "payment_expired",
        "payment_canceled",
        "payment_voided",
        "payment_void_declined",
        "payment_captured",
        "payment_capture_declined",
        "payment_capture_pending",
        "payment_refunded",
        "payment_refund_declined",
        "payment_refund_pending"
    }
};

try
{
    WebhookResponse response = api.WebhooksClient().PatchWebhook("webhook_id", request).Result;
}
catch (CheckoutApiException e)
{
    // API error
    string requestId = e.RequestId;
    var statusCode = e.HttpStatusCode;
    IDictionary<string, object> errorDetails = e.ErrorDetails;
}
catch (CheckoutArgumentException e)
{
    // Bad arguments
}
catch (CheckoutAuthorizationException e)
{
    // Invalid authorization
}

File: abc_spec/code_samples/Python/webhooks@{id}/patch.py
import checkout_sdk
from checkout_sdk.checkout_sdk import CheckoutSdk
from checkout_sdk.environment import Environment
from checkout_sdk.exception import CheckoutApiException, CheckoutArgumentException, CheckoutAuthorizationException
from checkout_sdk.webhooks.webhooks import WebhookRequest, WebhookContentType

api = CheckoutSdk.builder() \\
    .previous() \\
    .secret_key('secret_key') \\
    .environment(Environment.sandbox()) \\
    .build()
# or Environment.production()

request = WebhookRequest()
request.url = "https://example.com/webhooks"
request.active = True
request.content_type = WebhookContentType.JSON
request.event_types = ["payment_approved", "payment_pending", "payment_declined", "payment_expired", "payment_canceled", \\
   "payment_voided", "payment_void_declined", "payment_captured", "payment_capture_declined", "payment_capture_pending", \\
   "payment_refunded", "payment_refund_declined", "payment_refund_pending"]
request.headers = {
    "authorization": "1234"
}

try:
    response = api.webhooks.patch_webhook("webhook_id", request)
except CheckoutApiException as err:
    # API error
    request_id = err.request_id
    status_code = err.http_status_code
    error_details = err.error_details
    http_response = err.http_response
except CheckoutArgumentException as err:
# Bad arguments

except CheckoutAuthorizationException as err:
# Invalid authorization